### ISSUE TITLE
Request for support of rotating ID Token algorithm

### ISSUE THREAD

**Issue by dutta-arnab1** [auth0/express-openid-connect#616]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/express-openid-connect#readme), [Examples](https://github.com/auth0/express-openid-connect/blob/master/EXAMPLES.md), and [FAQ](https://github.com/auth0/express-openid-connect/blob/master/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/express-openid-connect/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/express-openid-connect/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

We use `express-openid-connect` library for performing `Authorization Code` flow login from a Node.js Express server. The purpose of this server is to enforce the user to be authenticated before we serve the actual web application.

A recent change in the private hosted organization level IDP had changed the default ID Token algorithm from `RS256` to `PS256`. We could resolve the problem just by loading the discovery document and set the first value in the `id_token_signing_alg_values_supported` array from discovery document into the config of `express-openid-connect -> auth middleware -> idTokenSigningAlg`.

However, the IDP team has announced that they would be supporting multiple ID Token algorithm and would rotate between them. So, the ID Token that would get generated could be signed with one of few defined algorithms.

I could not find a way to make it work with this library.
- `idTokenSigningAlg` receives only one algorithm.
- There is no option within the library to get the `kid` and `algorithm` out of JWT header and validate it against JWKS endpoint.
- Any option to pass on custom token verification and not perform the default JWT verification.

### Describe the ideal solution

Either of the three options mentioned would be nice to have.
- `idTokenSigningAlg` supporting an array of algorithms to be configured.
- Option within the library to get the `kid` and `algorithm` out of JWT header and validate it against JWKS endpoint.
- Any option to pass on custom token verification and not perform the default JWT verification.

**Comment by madaster97** [comment]
For your first option (supporting config of multiple algorithms), the library ([openid-client](https://github.com/panva/node-openid-client)) we [provide the config](https://github.com/auth0/express-openid-connect/blob/93a4fb0690532ba419d88e9b013220ba10d6a5d3/lib/client.js#L106) to also doesn't support multiple values. It may be worth creating an issue with that library as well.

The alternative to that would be complicated: supporting multiple instances of the library (clients) in this SDK. That sounds like a worse solution than having openid-client implement it.

**Comment by dutta-arnab1** [comment]
I agree, the alternative is not an option at all. So unless `openid-client` provides that option, there's nothing much we can do here.

Thanks @madaster97 for looking into this.
